packages/block-editor/src/components/block-toolbar/shuffle.js

```diff
@@ -34,7 +34,10 @@ export default function Shuffle( { clientId, as = Container } ) {
 			const _categories = attributes?.metadata?.categories || EMPTY_ARRAY;
 			const _patternName = attributes?.metadata?.patternName;
 			const rootBlock = getBlockRootClientId( clientId );
-			const _patterns = __experimentalGetAllowedPatterns( rootBlock );
+			const _patterns =
+				_categories.length > 0
+					? __experimentalGetAllowedPatterns( rootBlock )
```

Comment:
It's worth adding a comment why we're doing this, either here or wherever we consume the `patterns`. 

A comment can help in the future if someone is trying to use `patterns` for something not related to the `categories`, and they might be confused as to why `patterns` is empty sometimes.

Reply:
Good idea. I added to the comment.

We should also try to improve the `__experimentalGetAllowedPatterns` performance.